provider.go

```diff
@@ -17,6 +21,10 @@ type Provider struct {
 	APIToken  string `json:"api_token,omitempty"`  // API token with Zone.DNS:Write (can be scoped to single Zone if ZoneToken is also provided)
 	ZoneToken string `json:"zone_token,omitempty"` // Optional Zone:Read token (global scope)
 
+	// HTTPClient is the client used to communicate with Cloudflare.
+	// If nil, a default client will be used.
+	HTTPClient HTTPClient
```

Comment:
This should have a `json` struct tag to be consistent with the format of the other fields (we use `snake_case`), but also, I wonder if this should be a concrete type that can be serializable. OR, omit it from JSON entirely (`json:"-"`).

Reply:
@mholt thanks for reviewing this.

I would be ok with accepting a `*http.Client` but accepting the interface is preferable, letting anyone pass a generic client if desired.

I would be partial to adding this field to `Provider` tagging with omit  (`json:"-"`). I don't think it makes sense to serialize this field. 

I will make changes to just add the omit tag, and leave it at that for now, unless you have other feedback.